src/mscorlib/shared/System/IO/Path.Unix.cs

```diff
@@ -186,7 +191,15 @@ public static string GetPathRoot(string path)
             if (PathInternal.IsEffectivelyEmpty(path))
                 throw new ArgumentException(SR.Arg_PathEmpty, nameof(path));
 
-            return IsPathRooted(path) ? PathInternal.DirectorySeparatorCharAsString : String.Empty;
+            return new string(GetPathRoot(path.AsReadOnlySpan()));
```

Comment:
This should _never_ allocate. You don't want to call the other implementation in this case. The logic is too simple to have value calling the span overload and trying to figure out whether or not you got `/` as a result.